pkg/adsc/delta.go

```diff
@@ -487,6 +506,15 @@ func (c *Client) handleDeltaResponse(d *discovery.DeltaDiscoveryResponse) error
 	return nil
 }
 
+func (c *Client) markReceived(k resourceKey) {
+	if c.pendingWatches.Contains(k) {
+		c.pendingWatches.Delete(k)
+		if c.pendingWatches.Len() == 0 {
+			close(c.synced)
```

Comment:
how do you prevent double close?

Reply:
Think first this is closed, but at another time, `initWatch` is called again

Reply:
initWatch can never be called after it starts, it's something done during creation